### PR TITLE
Design 5 test scenarios

### PR DESCRIPTION
**Pull request by anciuvaga**

### CODE REVIEW

File: src/test/java/stepDefinition/HomePageStepDef.java [hunk 1]
```diff
@@ -17,4 +19,14 @@ public void userIsOnHomePage() {
         assertThat("Page is displayed", webDriver.getCurrentUrl(), is(getPage(getPageObjectName(HomePage.class)).getPageUrl()));
```
Comment (IulianCimpoes) [inline src/test/java/stepDefinition/HomePageStepDef.java:20]: assert message

File: src/test/resources/features/WishlistFunctionality.feature [hunk 2]
```diff
@@ -7,29 +7,27 @@ Feature: WishLish functionality
...
-#    When user checks 'addtocart' checkbox
-#    And user presses 'addtocartbutton' button
-#    Then selected 'Digital Storm VANQUISH 3 Custom Performance PC' from Wishlist is added successfully to ShoppingCart
+  @aciuvaga
+  Scenario: Add a product to WishList
+    Then product 'Nokia Lumia 1020' is displayed in the WishList
+
+  @aciuvaga
+  Scenario: Change product quantity in the WishList
+    When user press on 'qtyInput' Field
```
Comment (IulianCimpoes) [inline src/test/resources/features/WishlistFunctionality.feature]: I can't understand how user press the quantity filed if he don't choose any product on page and the same for the next 2 scenarios

File: src/main/java/pages/CategoryPage.java [hunk 3]
```diff
@@ -8,12 +8,15 @@
 @Getter
 public class CategoryPage extends PageObject {
 
-    @FindBy(css = " .category-grid.sub-category-grid>.item-grid")
-    private WebElement subCategory;
-
     //TODO: add required url for selected category
     private String pageUrl = "";
 
+    @FindBy( css= " .category-grid.sub-category-grid>.item-grid")
```
Comment (vdones) [inline src/main/java/pages/CategoryPage.java]: redundant WebElement

File: src/main/java/pages/HomePage.java [hunk 4]
```diff
@@ -1,20 +1,23 @@
...
 
-import lombok.Getter;
 
 @Getter
 public class HomePage extends PageObject {
 
     private String pageUrl = "https://demo.nopcommerce.com/";
 
+    @FindBy(xpath = "//ul[@class='top-menu notmobile']//a[contains(text(),'Electronics')]")
+    private WebElement category;
```
Comment (vdones) [inline src/main/java/pages/HomePage.java]: redundant WebElement

File: src/main/java/pages/ProductPage.java [hunk 5]
```diff
@@ -1,30 +1,44 @@
...
     private WebElement addToCartButton;
 
-    @FindBy(xpath = "//input[@id='add-to-wishlist-button-19']")
-    private WebElement addToWishButton;
+    @FindBy( css = " .add-to-wishlist input")
+    private WebElement addToWishlistButton;
 
-    @FindBy(xpath = "//span[@class='cart-label']")
+    @FindBy( xpath = "//span[@class='cart-label']")
     private WebElement ShoppingCartButton;
```
Comment (vdones) [inline src/main/java/pages/ProductPage.java]: code convention **shoppingCartButton**

File: src/main/java/pages/ProductPage.java [hunk 6]
```diff
@@ -1,30 +1,44 @@
...
 
-    @FindBy(xpath = "//span[@class='wishlist-label']")
+    @FindBy( xpath = "//span[@class='wishlist-label']")
     private WebElement WishListButton;
```
Comment (vdones) [inline src/main/java/pages/ProductPage.java]: code convention **wishListButton**

File: src/main/java/pages/ProductPage.java [hunk 7]
```diff
@@ -1,30 +1,44 @@
...
 
-    public ProductPage(final WebDriver driver) {
+    @FindBy( xpath = "//a[contains(text(),'wishlist')]")
+    private WebElement WishListLink;
```
Comment (vdones) [inline src/main/java/pages/ProductPage.java]: code convention **wishListLink**

File: src/main/java/pages/ProductPage.java [hunk 8]
```diff
@@ -1,30 +1,44 @@
...
+    private WebElement alertMessage;
+
+    @FindBy(css = "span.close")
+    private WebElement closeAlertMessage;
+
+    @FindBy( css = " .product-name > h1")
+    private WebElement productName;
+
+    @FindBy(css = " .header .header-upper .header-links-wrapper .header-links")
+    private WebElement headerForm;
```
Comment (vdones) [inline src/main/java/pages/ProductPage.java]: recheck webElements from this page
there are some elements which should be placed at abstractPage

File: src/test/java/stepDefinition/GenericsStepDef.java [hunk 9]
```diff
@@ -26,11 +26,17 @@ public void pageIsDisplayed(String pageName) {
...
     @When("^user press on '(.*)' (Button|Field|Checkbox)$")
     public void userPressOnButton(String elementName, String extension) {
-        implicitlyWait(5);
-        waitUntilElementIsClickable(getElementByName(elementName.concat(extension)));
+        waitVisibility(getElementByName(elementName.concat(extension)));
         clickOnElement(getElementByName(elementName.concat(extension)));
     }
 
+    @When("^user clicks on '(.*)' category from category bar$")
+    public void clickOnCategory(String categoryName) {
```
Comment (vdones) [inline src/test/java/stepDefinition/GenericsStepDef.java]: you added generic method for navigation, aren't you?

**Review by vdones — CHANGES_REQUESTED** [review]

File: src/test/resources/features/WishlistFunctionality.feature (hunk 2 above)
Comment (anciuvaga) [reply src/test/resources/features/WishlistFunctionality.feature]: I use background, and those steps are shared for all scenarios

File: src/main/java/pages/CategoryPage.java (hunk 3 above)
Comment (anciuvaga) [reply src/main/java/pages/CategoryPage.java]: Done

File: src/test/java/stepDefinition/GenericsStepDef.java (hunk 9 above)
Comment (anciuvaga) [reply src/test/java/stepDefinition/GenericsStepDef.java]: Done

File: src/main/java/pages/ProductPage.java (hunk 8 above)
Comment (anciuvaga) [reply src/main/java/pages/ProductPage.java]: Done

File: src/test/java/stepDefinition/HomePageStepDef.java (hunk 1 above)
Comment (anciuvaga) [reply src/test/java/stepDefinition/HomePageStepDef.java:20]: done

**Review by amiordan — APPROVED** [review]

**Review by MaximRaicov — APPROVED** [review]

**Pull request merged**